src/EfuncMsg.chpl

```diff
@@ -300,6 +420,390 @@ module EfuncMsg
+                    when "atan2" {
+                        var a = [(n) in zip(en.a)] atan2(n,val);
+                        st.addEntry(rname, new shared SymEntry(a));
+                    }
+                } 
+            }
+            when (DType.Float64, DType.UInt64) {
+                var en = toSymEntry(num, real);
+                var val = msgArgs.get("scalar").getUIntValue();
+                select efunc {
+                    when "atan2" {
+                        var a = [(n) in zip(en.a)] atan2(n,val);
+                        st.addEntry(rname, new shared SymEntry(a));
+                    }
+                } 
+            }
+            when (DType.Float64, DType.Float64) {
+                var en = toSymEntry(num, real);
+                var val = msgArgs.get("scalar").getRealValue();
+                select efunc {
+                    when "atan2" {
+                        var a = [(n) in zip(en.a)] atan2(n,val);
+                        st.addEntry(rname, new shared SymEntry(a));
+                    }
+                } 
+            }
+            otherwise {
+                var errorMsg = notImplementedError(pn,efunc,num.dtype,dtype);
+                eLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg); 
+                return new MsgTuple(errorMsg, MsgType.ERROR);            
+            }
+        }
+        repMsg = "created " + st.attrib(rname);
+        eLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg); 
+        return new MsgTuple(repMsg, MsgType.NORMAL); 
+    }
+
+    /*
+    vector = efunc(scalar, vector)
+
+    :arg reqMsg: request containing (cmd,efunc,scalar,denom,dtype)
+    :type reqMsg: string 
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: (MsgTuple)
+    :throws: `UndefinedSymbolError(name)`
+    */
+    proc efunc2svMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {
+        param pn = Reflection.getRoutineName();
+        var repMsg: string; // response message
+        var efunc = msgArgs.getValueOf("func");
+        var rname = st.nextName();
+        var dtype = str2dtype(msgArgs.getValueOf("dtype"));
+
+        var name = msgArgs.getValueOf("denom");
+        var denom: borrowed GenSymEntry = getGenericTypedArrayEntry(name, st);
+
+        eLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+            "cmd: %s efunc: %s scalar: %s dtype: %s name: %s rname: %s".format(
+             cmd,efunc,msgArgs.getValueOf("scalar"),dtype,name,rname));
+
+        select (dtype, denom.dtype) {
+            when (DType.Int64, DType.Int64) {
+                var val = msgArgs.get("scalar").getIntValue();
+                var ed = toSymEntry(denom, int);
+                select efunc {
+                    when "atan2" {
+                        var a = [(n) in zip(ed.a)] atan2(val,n);
+                        st.addEntry(rname, new shared SymEntry(a));
```

Comment:
you don't have to do zippered iteration if there's only one item. Also we should a `ref`
```suggestion
                var ed = toSymEntry(denom, int);
                ref eda = ed.a;
                select efunc {
                    when "atan2" {
                        var a = [d in eda] atan2(val,d);
                        st.addEntry(rname, new shared SymEntry(a));
```
Also since we're applying this to every element i wonder how this compares to doing the forall implicitly using promotion
```chapel
var ed = toSymEntry(denom, int);
ref eda = ed.a;
select efunc {
    when "atan2" {
        st.addEntry(rname, new shared SymEntry(atan2(val, eda)));
    }
}
```

Reply:
Updated. That makes sense, the zip is left over from the efunc3.